store/generated_key/memstore/v2/memstore.go

```diff
@@ -0,0 +1,219 @@
+	}
+
+	x := dataCommitment.X.BigInt(&big.Int{})
+	y := dataCommitment.Y.BigInt(&big.Int{})
+
+	g1CommitPoint := cert_verifier_binding.BN254G1Point{
+		X: x,
+		Y: y,
+	}
+
+	pseudoRandomBlobInclusionInfo := cert_verifier_binding.BlobInclusionInfo{
+		BlobCertificate: cert_verifier_binding.BlobCertificate{
+			BlobHeader: cert_verifier_binding.BlobHeaderV2{
+				Version:       0,                            // only supported version as of now
+				QuorumNumbers: []byte{byte(0x0), byte(0x1)}, // quorum 0 && quorum 1
+				Commitment: cert_verifier_binding.BlobCommitment{
+					LengthCommitment: cert_verifier_binding.BN254G2Point{
+						X: [2]*big.Int{randInt(1000), randInt(1000)},
+						Y: [2]*big.Int{randInt(1000), randInt(1000)},
+					},
+					LengthProof: cert_verifier_binding.BN254G2Point{
+						X: [2]*big.Int{randInt(1), randInt(1)},
+						Y: [2]*big.Int{randInt(1), randInt(1)},
+					},
+					Commitment: g1CommitPoint,
+					Length:     uint32(len(encodedVal)),
+				},
+				PaymentHeaderHash: [32]byte(randomBytes(32)),
+			},
+			Signature: randomBytes(48), // 384 bits
+			RelayKeys: []uint32{randUint32(), randUint32()},
+		},
+		BlobIndex:      uint32(randInt(1_000).Uint64()),
+		InclusionProof: randomBytes(128),
+	}
+
+	randomBatchHeader := cert_verifier_binding.BatchHeaderV2{
+		BatchRoot:            [32]byte(randomBytes(32)),
+		ReferenceBlockNumber: randUint32(),
+	}
+
+	randomNonSignerStakesAndSigs := cert_verifier_binding.NonSignerStakesAndSignature{
+		NonSignerQuorumBitmapIndices: []uint32{randUint32(), randUint32()},
+		NonSignerPubkeys: []cert_verifier_binding.BN254G1Point{
+			cert_verifier_binding.BN254G1Point{
+				X: randInt(1000),
+				Y: randInt(1000),
+			},
+		},
+		QuorumApks: []cert_verifier_binding.BN254G1Point{
+			cert_verifier_binding.BN254G1Point{
+				X: randInt(1000),
+				Y: randInt(1000),
+			},
+		},
+		ApkG2: cert_verifier_binding.BN254G2Point{
+			X: [2]*big.Int{randInt(1000), randInt(10000)},
+			Y: [2]*big.Int{randInt(1000), randInt(1000)},
+		},
+		QuorumApkIndices:  []uint32{randUint32(), randUint32()},
+		TotalStakeIndices: []uint32{randUint32(), randUint32(), randUint32()},
+		NonSignerStakeIndices: [][]uint32{
+			[]uint32{randUint32(), randUint32()},
+			[]uint32{randUint32(), randUint32()},
+		},
+		Sigma: cert_verifier_binding.BN254G1Point{
+			X: randInt(1000),
+			Y: randInt(1000),
+		},
+	}
+
+	artificialV2Cert := verification.EigenDACert{
+		BlobInclusionInfo:           pseudoRandomBlobInclusionInfo,
+		BatchHeader:                 randomBatchHeader,
+		NonSignerStakesAndSignature: randomNonSignerStakesAndSigs,
+	}
+
+	b, err := json.Marshal(artificialV2Cert)
+	if err != nil {
+		return nil, fmt.Errorf("unmarshal V2 cert: %w", err)
+	}
+
+	err = e.InsertEphemeralEntry(crypto.Keccak256Hash(b).Bytes(), encodedVal)
+	if err != nil { // don't wrap here so api.ErrorFailover{} isn't modified
+		return nil, err
+	}
+
+	certBytes, err := rlp.EncodeToBytes(artificialV2Cert)
+	if err != nil {
```

Comment:
should we not be using the BlobKey() function defined on cert in core repo to reuse the same hashing logic

Reply:
Yeah I'd rather not since the blob key only attests to the correctness of a few fields vs the entire cert. Ideally we should be able to verify from the rollup fork that the same certificate when dispersed by a batcher is the same one used in a derivation pipeline. Since we hash the entire cert for secondary storage insertion/lookup it gives us better holistic guarantees

Reply:
ahh good point!